processor/tailsamplingprocessor/internal/cache/types.go

```diff
@@ -0,0 +1,17 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package cache // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor/internal/cache"
+
+import "go.opentelemetry.io/collector/pdata/pcommon"
+
+// Cache is a cache using a pcommon.TraceID as the key and any generic type as the value.
+type Cache[V any] interface {
+	// Get returns the value for the given id, and a boolean to indicate whether the key was found.
+	// If the key is not present, the zero value is returned.
```

Comment:
I missed this -- is this because of the usage of generics there, or is there a different reason?

Reply:
You can't return `nil` when there's a generic type, so you return the zero value `var v V`. I think that's what you're asking.

Reply:
yes, thank you!